staging/src/k8s.io/kubectl/pkg/cmd/set/set_image.go

```diff
@@ -88,6 +88,11 @@ var (
 		kubectl set image -f path/to/file.yaml nginx=nginx:1.9.1 --local -o yaml`)
 )
 
+var (
+	imageResolver = resolveImageFunc
```

Comment:
Move here the current `ImageResolver` type and rename it to `ImageResolverFunc` so it matches our approach from https://github.com/kubernetes/kubernetes/blob/489fb9bee3f626b3eeb120a5af89ad8c2b2f1c20/staging/src/k8s.io/kubectl/pkg/polymorphichelpers/interface.go#L31-L34 so you should get:
```go
// ImageResolverFunc is a func that receives an image name, and
// resolves it to an appropriate / compatible image name.
// Adds flexibility for future image resolving methods.
type ImageResolverFunc func(in string) (string, error)

// ImageResolverFn gives a way to easily override the function for unit testing if needed.
var ImageResolverFn ImageResolverFunc = resolveImageFunc
```